backend/server.py

```diff
@@ -787,21 +801,25 @@ def delete_user(user_id):
         tuple: A tuple containing a JSON response and a status code.
     """
     # Check connection to the database
-    try:
-        connect_to_db()
-    except Exception as e:
-        print("Error connecting to the database:", str(e))
-        return (
-            jsonify({"error": "Database connection error"}),
-            status.HTTP_500_INTERNAL_SERVER_ERROR,
-        )
+    try_connect_to_db()
 
     try:
         # Connect to the database
         db = firestore.client()
 
-        # Delete the post from the 'posts' collection
-        db.collection("users").document(user_id).delete()
+        # Get the user reference
+        user_ref = db.collection("users").document(user_id)
+
+        # Get the user data
+        user_data = user_ref.get().to_dict()
+
+        # Delete all posts from the user

```

Comment:
This does delete the user but I'm not sure if the references are still there for the other user's post that have references connected to this user.